library/core/registry/src/main/java/org/quiltmc/qsl/registry/api/dynamic/DynamicRegistryFlag.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 The Quilt Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.api.dynamic;
+
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.registry.impl.dynamic.DynamicMetaRegistryImpl;
+import org.quiltmc.qsl.registry.impl.dynamic.DynamicRegistryFlagManager;
+
+/**
+ * Flags that can be set on dynamic registries to define their behavior.
+ * <p>All flags default to being disabled/turned off, and can be enabled or disabled using static methods in this class.
```

Comment:
"can be enabled or disabled"? it could be "can be toggled"! :p
it does help reducing the redundancy there;

```suggestion
 * <p>
 * All flags default to being turned off, and can be toggled using static methods in this class.
```

Reply:
The wording here is actually wrong, since they can only be enabled now. Going to fix it in a commit after the upcoming one that adds most of your other style suggestions

Reply:
Should be good now